### PR TITLE
refactor/3052 refactor to avoid having too many files in root

### LINKED ISSUE #3052: refactor `.` to avoid having too many files in root
**Is your feature request related to a problem? Please describe.**
There are way too many files in root, which could be moved to thing like '.github`.

**Describe the solution you'd like**
N.A.

**Describe alternatives you've considered**
N.A.

**Additional context**
N.A.

### PR DESCRIPTION
**Pull request by davidberenstein1957**
# Description

Moved some stuff
- docker-related files to `docker` directory
- `scripts` directory to `docker` directory 
- general files to `.github` 

Closes #3052

**Type of change**

- [X] Refactor (change restructuring the codebase without changing functionality)

**How Has This Been Tested**

N.A.

**Checklist**

N.A.

### CODE REVIEW

**Comment by davidberenstein1957** [comment]
@frascuchon we don't need to forget to set the read the docs config path to `docs/_source/.readthedocs.yaml`

**Comment by davidberenstein1957** [comment]
I tried to move the quickstart docker files to the `.github` folder too but it seemed that the build pipeline do not have access to `.github`.

File: .github/MANIFEST.in
Comment (frascuchon) [inline .github/MANIFEST.in:1]: This file is used on package build. Not sure if we can move it from root without changes in pyproject.tml.

File: docs/_source/.readthedocs.yaml
Comment (frascuchon) [inline docs/_source/.readthedocs.yaml:1]: This change will affect the docs build. See [the docs](https://docs.readthedocs.io/en/stable/config-file/v2.html#configuration-file-v2), but this file must be placed in the root of the project.

Comment (davidberenstein1957) [reply docs/_source/.readthedocs.yaml:1]: @frascuchon  We can reference it in a nested directory too:) see my comment above.

File: .github/MANIFEST.in
Comment (davidberenstein1957) [reply .github/MANIFEST.in:1]: The build seems to run without warnings but I can check to make sure.

File: .github/.codecov.yml
Comment (frascuchon) [inline .github/.codecov.yml:1]: Same here, this file should placed in the root folder

File: .github/MANIFEST.in
Comment (davidberenstein1957) [reply .github/MANIFEST.in:1]: `pyproject.tml` does not seem to reference `MANIFEST.in`

File: .github/.codecov.yml
Comment (davidberenstein1957) [reply .github/.codecov.yml:1]: I believe this should work in .github too https://github.com/codecov/codecov-ruby/issues/24#issuecomment-644772335

File: docs/_source/.readthedocs.yaml
Comment (frascuchon) [reply docs/_source/.readthedocs.yaml:1]: It looks like an experimental feature. I would prefer to keep the default value We never know with rtd builds... ;-)

Screenshot 2023-06-01 at 22 01 20

**Comment by frascuchon** [comment]
As an easy step, what we can do is move all docker stuff (composes and dockerfiles) to a docker folder. Also, we can remove the commitlint configuration (since we're managing the changelog manually) and maybe the simpler ones to the .github folder too (README.md, CODE_OF_CONDUCT.md, LICENCE...)

**Comment by davidberenstein1957** [comment]
@frascuchon can you take a look at this PR?

File: docker/docker-compose.yaml
Comment (frascuchon) [inline docker/docker-compose.yaml:1]: We should change also the refs to this file in docs https://docs.argilla.io/en/latest/getting_started/installation/deployments/docker_compose.html#launching-argilla-and-elasticsearch-with-docker-compose

Comment (frascuchon) [reply docker/docker-compose.yaml:1]: Also here https://docs.argilla.io/en/latest/getting_started/installation/deployments/from_development.html#install-from-develop

File: docker/docker-compose.opensearch.yaml
Comment (frascuchon) [inline docker/docker-compose.opensearch.yaml:1]: I think we have also references to these files in docs?

Comment (davidberenstein1957) [reply docker/docker-compose.opensearch.yaml:1]: Yes, I had already updated those and the ones from `docker-compose.elasticsearch.yml`, but forgot about the docker-compose.

File: docker/docker-compose.yaml
Comment (davidberenstein1957) [reply docker/docker-compose.yaml:1]: Thanks, I forgot those.

**Comment by davidberenstein1957** [comment]
@frascuchon I changed the branch to `main` to avoid merge conflict with develop during the coming merges, and to avoid redirecting the `.yml` files to a non existend `/main/docker/*.yml`

**Review by frascuchon — COMMENTED** [review]
Regarding this [doc section](https://docs.github.com/en/communities), I'm not sure if we should move all those files into the dot-github folder. 

It looks like this folder is used for templates and workflow definitions. It's a bit weird for me to find README.md or CHANGELOG.md there, instead of the root one. Even more for the conda environment `environment_dev.yml`  or the coverage setup (`.codecov.yml`)

I would say:

1. keep common repo files in the root folder (README.md CHANGELOG, coverage setup, conda env) 
2. Move al docker stuff into a docker folder and review the required scripts
3. Remove unused files as the commitlint.config.js file or some obsolete script (I can take a look here)

Also, I think we should point this PR to the `develop` branch, right?

**Comment by frascuchon** [comment]
I'm not sure if this checklist can be affected with these changes https://github.com/argilla-io/argilla/community

File: .github/workflows/package.yml
Comment (frascuchon) [inline .github/workflows/package.yml]:
```suggestion
            readme: README.md
```

**Comment by davidberenstein1957** [comment]
@frascuchon  can this be merged?

File: docker/Dockerfile
Comment (frascuchon) [inline docker/Dockerfile]:
```suggestion
CMD /bin/bash /start_argilla_server.sh
```

**Review by frascuchon — APPROVED** [review]
It's working. Let's wait until the release branch creation before merging it, just to not disturb the release workflow

**Comment by davidberenstein1957** [comment]
@frascuchon can you show me tomorrow how to resolve these conflicts?

**Pull request merged**
